src/utils/getUrlByRequestOptions.test.ts

```diff
@@ -105,10 +105,15 @@ it('resolves hostname to localhost if none provided', () => {
   expect(getUrlByRequestOptions({}).hostname).toBe('localhost')
 })
 
-it('supports "hostname" instead of "host" and "port"', () => {
+it('resolves host to localhost if none provided', () => {
+  expect(getUrlByRequestOptions({}).host).toBe('localhost')
+})
+
+it('supports "hostname" and "port"', () => {
   const options: RequestOptions = {
     protocol: 'https:',
-    hostname: '127.0.0.1:1234',
```

Comment:
I'd be extremely careful around changes like this. 

Yes, this is technically right—`hostname` does not include port. But in practice, I've seen a lot of libraries misuse Node.js. I fear this was one of those cases.

Reply:
@kettanaito, can you give an example? Because it's invalid and Node throws an error immediately:

![image](https://github.com/mswjs/interceptors/assets/11459632/9eafd57c-f254-44ce-acb7-26e0f62ae38c)

And this is working: 
![image](https://github.com/mswjs/interceptors/assets/11459632/60d4a921-5663-4859-b5ee-6d463028dfe6)

Reply:
That's a good point! Let's remove this old logic and if any issues arise, tackle them separately. I have no notion of supporting hacky tools, and if you use `http.request()` incorrectly, you are hacky.